src/PimApi/FetchOrCreateAppCatalogHandler.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\PimApi;
+
+use App\Storage\CatalogIdStorageInterface;
+use App\Storage\PimURLStorageInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class FetchOrCreateAppCatalogHandler
```

Comment:
Could you name it differently ? it's not an handler, it's kinda confusing.

What do you think of `FetchOrCreateDefaultDemoCatalogQuery` and move it to `Query` ?

Also, the execute method should return the catalog id, not store it.
it should be passed to the storage by the controller.